modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth/dcrm/OAuthDCRMReadTestCase.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * you may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.identity.integration.test.oauth.dcrm;
+
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.JSONValue;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.identity.integration.common.utils.ISIntegrationTest;
+import org.wso2.identity.integration.test.oauth.dcrm.bean.ServiceProvider;
+import org.wso2.identity.integration.test.oauth.dcrm.util.OAuthDCRMConstants;
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class OAuthDCRMReadTestCase extends ISIntegrationTest {
+
+    private HttpClient client;
+    private ServiceProviderRegister serviceProviderRegister;
+    private ServiceProvider serviceProvider;
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+        client = new DefaultHttpClient();
+        serviceProviderRegister = new ServiceProviderRegister();
+
+        JSONObject object = new JSONObject();
+        object.put(OAuthDCRMConstants.CLIENT_NAME, "ReadApp");
+        object.put(OAuthDCRMConstants.GRANT_TYPES, "implicit");
+        object.put(OAuthDCRMConstants.REDIRECT_URIS, "http://ReadApp.com");
+
+        serviceProvider = serviceProviderRegister.register(object.toJSONString());
```

Comment:
We dont need the SP to be remain the Server after the test is completed. isnt it? If so add a @AfterClass and remove the SP added after the test execution .